source/Classes/1.DscResources/07.nxService.ps1

```diff
@@ -40,7 +40,7 @@ class nxService
         $currentState.State = $nxService.State
         $currentState.Controller = $this.Controller
 
-        if (-not $currentState)
```

Comment:
This check should happen before the info from the service is retrieved in the lines above. (It should be a check on the retrieved $nxService rather than $currentState too)
Also $currentState.Name is not a boolean so calling it as a boolean can have some unpredictable behavior.
This should be something like $null -eq $currentState.Name or some boolean expression like that